src/hashes.ts

```diff
@@ -22,9 +22,9 @@ const Sha512Half = makeClass({
     return bytes.slice(0, 32);
   },
   finish() {
-    return new Hash256(this.finish256());
+    return new coreTypes.Hash256(this.finish256());
   }
-});
+}, undefined);
```

Comment:
In JS, unspecified params are implicitly `undefined`. `makeClass` takes two parameters and the TypeScript compiler errors if we don't set both params.

Reply:
I guess this is probably out of scope, but this smells of something that is overly clever. 

Why are we making a class instead of just writing one?

Reply:
Library written before classes were introduced. I currently have a local branch that has been changed to use Javascript classes.